File: goblin-cache/src/main/java/org/goblinframework/cache/module/test/FlushCache.java
package org.goblinframework.cache.module.test;

import org.goblinframework.cache.core.cache.CacheSystem;

import java.lang.annotation.*;

@Documented
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface FlushCache {

  CacheSystem system();

  String connection();

}

File: goblinframework-cache/src/main/java/org/goblinframework/cache/core/annotation/FlushCache.kt
package org.goblinframework.cache.core.annotation

import org.goblinframework.cache.core.cache.CacheSystem

@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class FlushCache(val system: CacheSystem, val name: String)
